partcad/src/partcad/assembly.py

```
@@ -64,67 +58,63 @@ def add(
         # Keep part reference counter for bill-of-materials purposes
         child_item.ref_inc()
 
-        self.shape = None  # Invalidate if any
+        self._wrapped = None  # Invalidate if any
 
     def ref_inc(self):
         for child in self.children:
             child.item.ref_inc()
 
-    async def get_shape(self):
+    async def get_shape(self, ctx):
```

Ruff (0.8.2)

Missing return type annotation for public function `get_shape`

(ANN201)